myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -116,8 +114,21 @@ public void onEvent(ResourceOffersEvent event, long sequence,
                         driver.declineOffer(offer.getId());
                     }
                 }
-            } else {
-                offerLifecycleMgr.addOffers(offers);
+            }
+
+            for (Offer offer : offers) {
+              if (SchedulerUtils.isEligibleForFineGrainedScaling(offer.getHostname(), schedulerState)) {
```

Comment:
Potentially grabbing a resource twice?  Say there is a pending nm.zero task, it gets added then the offers list is iterated over again to increase the resources of the nm.zero task.  This might cause a few TASK_LOST messages but likely will not cause any real issues.